File: frontend/src/services/storageRackService.ts
import { apiClient } from '@/config/apiClient';
import type { 
  ApiResponse, 
  PageResponse, 
  StorageRackResponse, 
  StorageRackRequest, 
  TableParams 
} from '@/types';

export const storageRackService = {
  getAll: (params?: TableParams & { status?: string }): Promise<ApiResponse<PageResponse<StorageRackResponse>>> => {
    return apiClient.get('/storage-racks', { params });
  },

  getAvailable: (): Promise<ApiResponse<StorageRackResponse[]>> => {
    return apiClient.get('/storage-racks/available');
  },

  getById: (id: number): Promise<ApiResponse<StorageRackResponse>> => {
    return apiClient.get(`/storage-racks/${id}`);
  },

  create: (data: StorageRackRequest): Promise<ApiResponse<StorageRackResponse>> => {
    return apiClient.post('/storage-racks', data);
  },

  update: (id: number, data: StorageRackRequest): Promise<ApiResponse<StorageRackResponse>> => {
    return apiClient.put(`/storage-racks/${id}`, data);
  },

  delete: (id: number): Promise<ApiResponse<void>> => {
    return apiClient.delete(`/storage-racks/${id}`);
  },
};

File: backend/src/main/java/com/bubbleflow/service/StorageRackService.java
package com.bubbleflow.service;

import com.bubbleflow.dto.storage.request.StorageRackRequest;
import com.bubbleflow.dto.storage.response.StorageRackResponse;
import com.bubbleflow.dto.response.PageResponse;
import org.springframework.data.domain.Pageable;
import java.util.List;

public interface StorageRackService {
    PageResponse<StorageRackResponse> getAll(String search, String status, Pageable pageable);
    List<StorageRackResponse> getAvailableRacks();
    StorageRackResponse getById(Long id);
    StorageRackResponse create(StorageRackRequest request);
    StorageRackResponse update(Long id, StorageRackRequest request);
    void delete(Long id);
}
